3rdparty/snmalloc/CMakeLists.txt

```diff
@@ -1,7 +1,12 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
+check_submodule_not_empty("${PROJECT_SOURCE_DIR}/3rdparty/snmalloc/snmalloc")
 add_enclave_library(oesnmalloc OBJECT allocator.cpp)
+# Add explicit dependency to musl_includes and libcxx includes.
+# Specifying a dependency on oelibc_includes does not seem to work reliably
+# on all platforms.
+add_dependencies(oesnmalloc libcxx_includes musl_includes)
 
```

Comment:
If you add the following, then it should pull in the required compiler settings from snmalloc:
```suggestion

# Include snmalloc library
set(SNMALLOC_ONLY_HEADER_LIBRARY ON)
add_subdirectory(snmalloc EXCLUDE_FROM_ALL)
target_link_libraries(oesnmalloc PRIVATE snmalloc_lib)

```
This has the advantage that if we alter our dependencies then it should get updated here automatically.  However, as you are building in a constrained setting, we may specify things that don't work here.  Possible `target_link_library` should be `enclave_link_library` but I don't know you specifics well enough.

Reply:
Thanks for the suggestion. I have retained the current way since we are building in a constrained setting and would like to have precise control over the way snmalloc is built as part of the SDK.